src/com/amazon/ion/impl/_Private_Utils.java

```diff
@@ -879,6 +879,13 @@ public static IonStruct symtabTree(SymbolTable symtab, ValueFactory valueFactory
         } else {
             localSymbolTableAsStruct = (LocalSymbolTableAsStruct) new LocalSymbolTableAsStruct.Factory(valueFactory)
                     .newLocalSymtab(symtab.getSystemSymbolTable(), symtab.getImportedTables());
+            Iterator<String> localSymbolsIterator = symtab.iterateDeclaredSymbolNames();
+            while (localSymbolsIterator.hasNext()) {
+                String localSymbol = localSymbolsIterator.next();
+                if (localSymbol != null) {
+                    localSymbolTableAsStruct.intern(localSymbol);
+                }
+            }
```

Comment:
The fact that this was missing was a bug; the symbol table's local symbols were not added to the IonStruct representation. Nothing bad happened because this method was only called in cases where readers created `LocalSymbolTableAsStruct` instances. The incremental reader does not, so now this code path is hit. I will create a follow-up PR that makes the incremental reader lazily memoize the IonStruct representations of its local symbol tables, as this leads to a significant performance improvement in rare cases.

Reply:
Good find 👍